Post: Introducing latest aliases (#259)
readthedocs/website: content/posts/introducing-latest-aliases.md

> title: Introducing latest aliases for operating system and tools in the configuration file

Reviewer: The long title here breaks some of the blog, but might also sound clearer shortened a bit:

![image](https://github.com/readthedocs/website/assets/1140183/554db155-4390-45f2-ab75-772c3516ba7b)

```suggestion
title: New tool and operating system aliases
```

Author: What about "Introducing latest options for operating system and build tools"?

Reviewer: You could try that, the breadcrumbs were meant to fit on a single line. The page title in the breadcrumb should probably be truncated in the template too. But overall shorter titles will get reader attention easier too.

date: 2024-02-15

> category: changelog
> tags: configuration file

Reviewer 2: ```suggestion
category: Changelog
tags: configuration-file, config-file
```

authors: Santos Gallegos
status: published

While some projects benefit from the stability provided by using a specific version,
and updating when needed, others projects, especially those under active development,
prefer to use the latest versions available to take advantage of new features and improvements

> without having to update the configuration file every time a new version is available.

Author: If we want to keep it shorter, we can remove this paragraph.

Reviewer: I think it adds a good amount of background actually. I'd keep it, but I'd suggest breaking it up a bit more to keep it easy to read.

Reviewer 2: Could maybe use at least one period :)

Reviewer 2: ```suggestion
We generally recommend projects pin specific versions of their dependencies,
so that you have [reproducible builds](https://docs.readthedocs.io/en/stable/guides/reproducible-builds.html) across time.
However, some projects under active development prefer to use the latest versions available to take advantage of new features and improvements,
without having to update the configuration file every time a new version is available.
```

> While some projects benefit from the stability provided by using a specific version,
> and updating when needed, others projects, especially those under active development,
> prefer to use the latest versions available to take advantage of new features and improvements
> without having to update the configuration file every time a new version is available.

Reviewer: These are all just suggestions, mostly breaking up the paragraph a bit more. Feel free to cherry pick these edits:

```suggestion
While most projects benefit from the stability provided by using specific versions for system requirements, other projects, especially those under active development,
prefer to use the latest versions available. This allows these projects to take advantage of new features and improvements
without having to update their configuration file every time a new version is released.
```

Reviewer: Bleh, what is up with GitHub and these awful diffs? Am I doing something incorrect?

> We are happy to announce the support for using the latest version available of the

Reviewer: ```suggestion
We are happy to announce support for using the latest available version for the
```

[operating system](https://docs.readthedocs.io/en/stable/config-file/v2.html#build-os)

> and [tools](https://docs.readthedocs.io/en/stable/config-file/v2.html#build-tools) in the configuration file.

Reviewer: I usually call these `the build.os configuration option` or similar:

```suggestion
and [tools](https://docs.readthedocs.io/en/stable/config-file/v2.html#build-tools) configuration options.
```

> The following new options are available:

Reviewer: ```suggestion
The following new option values are available:
```

- `build.os`: `latest`
- `build.tools.python`: `latest`, `miniconda-latest`, `mambaforge-latest`
- `build.tools.nodejs`: `latest`
- `build.tools.ruby`: `latest`
- `build.tools.rust`: `latest`
- `build.tools.golang`: `latest`

> When using those options, **the latest version available on Read the Docs will be used,

Reviewer: ```suggestion
When using these values, **the latest version available on Read the Docs will be used,
```

which may not match the latest version officially released**.

> Check our documentation for the list of versions available for each option.

Reviewer: ```suggestion
[Check our documentation](#link) for the list of versions available for each option.
```

> Versions will be updated at least once every six months,

Reviewer: There is terminology overlap here with project versions, to differentiate, something like this perhaps:

```suggestion
New versions of system requirements will be updated at least once every six months.
```

> keep in mind that your builds may break unexpectedly if your project isn't compatible with the newest versions when they are updated on Read the Docs.

Reviewer: I'd break this to a new sentence

```suggestion
Keep in mind that your builds may break unexpectedly if your project isn't compatible with the newest versions when they are updated on Read the Docs.
```

> Versions will be updated at least once every six months,
> keep in mind that your builds may break unexpectedly if your project isn't compatible with the newest versions when they are updated on Read the Docs.

Reviewer 2: ```suggestion
Versions will be updated at least once every six months.
**Keep in mind that your builds may break unexpectedly if your project isn't compatible with the newest versions when they are updated on Read the Docs.**
```

> We recommend using a specific version for projects that require stability.

Reviewer: We could make this point stronger too, as we probably still want most projects using explicit versions. "We recommend most projects ..." or something.

> We recommend using a specific version for projects that require stability.

Reviewer 2: ```suggestion
We recommend using a specific versions of dependencies for projects that require stability.
```
